quickwit/quickwit-search/src/collector.rs

```diff
@@ -684,6 +730,8 @@ pub(crate) fn sort_by_from_request(search_request: &SearchRequest) -> SortByPair
     let to_sort_by_component = |field_name: &str, order| {
         if field_name == "_score" {
             SortByComponent::Score { order }
+        } else if field_name == "_shard_doc" || field_name == "_doc" {
+            SortByComponent::DocId { order }
```

Comment:
I don't think this is the place to have `_shard_doc`
If we want to support it, that should be in the Elasticsearch REST API handler.